Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift

```diff
@@ -65,7 +66,10 @@ class DefaultPublisher: Publisher {
          ablyService: AblyPublisherService,
          locationService: LocationService,
          routeProvider: RouteProvider,
-         trackableState: PublisherTrackableState = .init()) {
+         trackableState: PublisherTrackableState = DefaultTrackableState(),
+         skippedLocationsState: PublisherSkippedLocationsState = DefaultSkippedLocationsState()
```

Comment:
Isn't the trackable updates stored in `PublisherSkippedLocationsState` fit nicer as part of `PublisherTrackableState`? These are locations which are for the trackable. It is rightfully its state. Ably-java does it the same way as you though.

Reply:
All of this code is based on AAT Android (kotlin).
```kotlin
private inner class State(
        routingProfile: RoutingProfile,
        locationEngineResolution: Resolution
) {
    private var isDisposed: Boolean = false
    var isStopped: Boolean = false
    var locationEngineResolution: Resolution = locationEngineResolution
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    var isTracking: Boolean = false
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    val trackables: MutableSet<Trackable> = mutableSetOf()
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    val trackableStates: MutableMap<String, TrackableState> = mutableMapOf()
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    val trackableStateFlows: MutableMap<String, MutableStateFlow<TrackableState>> = mutableMapOf()
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    val lastChannelConnectionStateChanges: MutableMap<String, ConnectionStateChange> = mutableMapOf()
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    var lastConnectionStateChange: ConnectionStateChange = ConnectionStateChange(
        ConnectionState.OFFLINE, null
    )
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    val resolutions: MutableMap<String, Resolution> = mutableMapOf()
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    val lastSentEnhancedLocations: MutableMap<String, Location> = mutableMapOf()
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    val skippedEnhancedLocations: SkippedLocations = SkippedLocations()
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    var estimatedArrivalTimeInMilliseconds: Long? = null
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    var lastPublisherLocation: Location? = null
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    var currentDestination: Destination? = null
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    val subscribers: MutableMap<String, MutableSet<Subscriber>> = mutableMapOf()
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    val requests: MutableMap<String, MutableMap<Subscriber, Resolution>> = mutableMapOf()
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    var presenceData: PresenceData = PresenceData(ClientTypes.PUBLISHER)
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    var active: Trackable? = null
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
        set(value) {
            this@DefaultCorePublisher.active = value
            field = value
        }
    var routingProfile: RoutingProfile = routingProfile
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
        set(value) {
            this@DefaultCorePublisher.routingProfile = value
            field = value
        }
    val rawLocationChangedCommands: MutableList<(State) -> Unit> = mutableListOf()
        get() = if (isDisposed) throw PublisherStateDisposedException() else field
    val enhancedLocationsPublishingState: LocationsPublishingState = LocationsPublishingState()
        get() = if (isDisposed) throw PublisherStateDisposedException() else field

    fun dispose() {
        trackables.clear()
        trackableStates.clear()
        trackableStateFlows.clear()
        lastChannelConnectionStateChanges.clear()
        resolutions.clear()
        lastSentEnhancedLocations.clear()
        skippedEnhancedLocations.clearAll()
        estimatedArrivalTimeInMilliseconds = null
        active = null
        lastPublisherLocation = null
        currentDestination = null
        subscribers.clear()
        requests.clear()
        rawLocationChangedCommands.clear()
        enhancedLocationsPublishingState.clearAll()
        isDisposed = true
    }
}
```

Reply:
I already guessed you based it off AATA:
>  I think Ably-java does it the same way as you though.

I'm not sure if you agree or disagree with my first comment in this thread 😅?

Reply:
96b0db747bbf5f6938a0a3a18533c1a14b035154